### PR TITLE
Draft: Feat: Enable torch compile

### PR DESCRIPTION
**Pull request by jiemingz**
# What does this PR do ?

**Add a one line overview of what this PR aims to accomplish.**

# Issues
List issues that this PR closes ([syntax](https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)):

# Usage
* **You can potentially add a usage example below**

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](/NVIDIA/NeMo-RL/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you run the unit tests and functional tests locally? Visit our [Testing Guide](/NVIDIA/NeMo-RL/blob/main/docs/testing.md) for how to run tests
- [ ] Did you add or update any necessary documentation? Visit our [Document Development Guide](/NVIDIA/NeMo-RL/blob/main/docs/documentation.md) for how to write, build and test the docs.

# Additional Information
* ...

### CODE REVIEW

File: examples/configs/dpo.yaml
Comment (terrykong) [inline examples/configs/dpo.yaml:1]: Could you add this key to all the `configs/recipes`?

**Review by terrykong — COMMENTED** [review]
Is this possible to unit test?

**Comment by terrykong** [comment]
@jiemingz is the only thing blocking this PR the seq-packing change since we need static shapes for torch.compile?

**Comment by SahilJain314** [comment]
Dtensor sequence packing has been merged. @ahmadki to support max-padding packed sequences in DTensor to enable torch.compile (fixed seqlen).

**Comment by ahmadki** [comment]
> Dtensor sequence packing has been merged. @ahmadki to support max-padding packed sequences in DTensor to enable torch.compile (fixed seqlen).

tracking [here](https://github.com/NVIDIA-NeMo/RL/issues/740)

File: nemo_rl/models/policy/dtensor_policy_worker.py [hunk 1]
```diff
@@ -195,6 +196,9 @@ def __init__(
             custom_parallel_plan=self.cfg["dtensor_cfg"]["custom_parallel_plan"],
         )
 
+        if self.torch_compile:
+            self.model = torch.compile(model)
```
Comment (StrongerXi) [inline nemo_rl/models/policy/dtensor_policy_worker.py:200]: Could you try `model.compile()` instead? That should fix the `_orig_mod` issue. This is also the [recommended way](https://docs.pytorch.org/tutorials/intermediate/torch_compile_tutorial.html#best-practices-and-recommendations) of compiling a model now. We'll work on throwing warnings and publicizing to raise awareness on this.

**Comment by euronymous-aithal** [comment]
@yuki-97  can you please review this and take this forward. 
(QQ: @ffrujeri  and @joyang-nv  not sure if Automodel path already enables torch.compile if so we should close this )

**Comment by terrykong** [comment]
Closing since this is w.r.t. v1 and we are currently centralizing efforts around v2 . closing this and reassigned https://github.com/NVIDIA-NeMo/RL/issues/4 to @joyang-nv  for now
